File: platform/sdk/python/examples/t3_07_upload_10parts_big.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))
from xuanji_sdk_cloud import CloudClient

def main():
    client = CloudClient()
    bucket = "bigdata"
    key = "reports/2026/q2/large.csv"
    r0 = client.createMultipartUpload(bucket, key)
    if not r0["ok"]:
        raise RuntimeError("create failed")
    uid = r0["upload_id"]
    CHUNK = 64 * 1024
    parts = []
    total = 0
    for n in range(1, 11):
        data = bytearray([n & 0xFF] * CHUNK)
        for i in range(8):
            data[i] = (n + i) & 0xFF
        total += len(data)
        r = client.uploadPart(bucket, key, uid, n, bytes(data))
        if not r["ok"]:
            raise RuntimeError(f"part {n} upload failed")
        parts.append({"part_number": n, "etag": r["etag"]})
    if len(parts) != 10:
        raise RuntimeError(f"expected 10 parts got {len(parts)}")
    fin = client.completeMultipartUpload(bucket, key, uid, parts)
    if not fin["ok"]:
        raise RuntimeError("complete failed")
    obj = client.getObject(bucket, key)
    if not obj["found"]:
        raise RuntimeError("obj not found")
    data = obj["data"] if obj["data"] is not None else b""
    size = len(data)
    if size != total:
        raise RuntimeError(f"total size mismatch: expected={total} got={size}")
    if total != 10 * CHUNK:
        raise RuntimeError("CHUNK total mismatch")
    print(f"XJ-OK: t3_07_upload_10parts_big parts=10 total_bytes={total} etag={fin['etag']}")

if __name__ == "__main__":
    main()

File: platform/sdk/nodejs/examples/t3-07-upload-10parts-big.js
const { CloudClient } = require("../mox-sdk-cloud");

function main() {
  const client = new CloudClient();
  const bucket = "bigdata";
  const key = "reports/2026/q2/large.csv";
  const r0 = client.createMultipartUpload(bucket, key);
  if (!r0.ok) throw new Error("create failed");
  const uid = r0.upload_id;
  const CHUNK = 64 * 1024;
  const parts = [];
  let total = 0;
  for (let n = 1; n <= 10; n++) {
    const data = Buffer.alloc(CHUNK, n & 0xFF);
    for (let i = 0; i < 8; i++) data[i] = (n + i) & 0xFF;
    total += data.length;
    const r = client.uploadPart(bucket, key, uid, n, data);
    if (!r.ok) throw new Error("part " + n + " upload failed");
    parts.push({ part_number: n, etag: r.etag });
  }
  if (parts.length !== 10) throw new Error("expected 10 parts, got " + parts.length);
  const fin = client.completeMultipartUpload(bucket, key, uid, parts);
  if (!fin.ok) throw new Error("complete failed");
  const obj = client.getObject(bucket, key);
  if (!obj.found) throw new Error("obj not found");
  const size = (obj.data && obj.data.length) || 0;
  if (size !== total) throw new Error("total size mismatch: expected=" + total + " got=" + size);
  if (total !== 10 * CHUNK) throw new Error("CHUNK total mismatch");
  console.log("XJ-OK: t3-07-upload-10parts-big parts=10 total_bytes=" + total + " etag=" + fin.etag);
}

if (require.main === module) main();
module.exports = main;
